solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -756,6 +763,12 @@ public void load() {
         // is set to true for them
         // TODO: should this go behind some config?
         List<CoreDescriptor> additionalCoreDescriptors = discoverAdditionalCoreDescriptorsForSharedReplicas(cds);
+        if (!isSharedStoreEnabled() && discoveredSharedCollection) {
+          throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, 
```

Comment:
This doesn't actually stop the process from starting up and currently not sure of a good way to abort the process if it's misconfigured. There used to be a setting by that was deprecated a long time ago.